src/ocaml-config/ocaml_config.ml

```diff
@@ -55,20 +55,48 @@ end
 module Ccomp_type = struct
   type t =
     | Msvc
+    | Gcc
+    | Clang
     | Other of string
 
   let to_dyn =
     let open Dyn.Encoder in
     function
     | Msvc -> constr "Msvc" []
+    | Gcc -> constr "Gcc" []
+    | Clang -> constr "Clang" []
     | Other s -> constr "Other" [ string s ]
 
-  let of_string = function
-    | "msvc" -> Msvc
-    | s -> Other s
+  let readlinks bin =
+    let path = Env.(path initial) in
+    let rec aux l =
+      let path =
+        Bin.which ~path l
+        |> Option.map ~f:Path.to_absolute_filename
+        |> Option.value ~default:bin
+      in
+      try
+        let ll = Unix.readlink path in
+        if ll = l then
+          l
+        else
+          aux ll
```

Comment:
This won't handle loops - Linux does this by only allowing a maximum recursion depth of 8.